packages/Angular/Explorer/explorer-core/src/lib/resource-browser/resource-browser.component.ts

```diff
@@ -362,168 +444,90 @@ export class ResourceBrowserComponent {
+    if(!folder || !folder.CategoryEntity){
+      LogError("Folder or CategoryEntity is not set, cannot delete folder");
+      this.sharedService.CreateSimpleNotification("Unable to delete folder", "error", 2500);
+      return false;
+    }
 
-    // //the DB will throw an error if we attempt to delete a folder that has children
-    // //i.e. sub folders or resources
-    // //so the default behavior is to notify the user that the delete operation cannot
-    // //go through 
-    // const folderHasChildren: boolean = await this.doesFolderHaveChildren(folder.ID);
-    // if(folderHasChildren){
-    //   this.showNotification(`unable to delete folder ${folder.Name} because it has children`, "error");
-    //   return false;
-    // }
+    const hasChildren: boolean = await this.doesFolderHaveChildren(folder.ID);
+    if(hasChildren){
+      this.sharedService.CreateSimpleNotification(`Unable to delete Folder ${folder.Name} because it has children`, "error", 2500);
+      return false;
+    }
 
-    // this.showLoader = true;
-    // const md = new Metadata();
-    // let folderEntity: BaseEntity = await md.GetEntityObject<BaseEntity>(this.CategoryEntityName);
-    // let pkv: KeyValuePair = new KeyValuePair();
-    // pkv.FieldName = "ID";
-    // pkv.Value = folder.ID;
-    // let compositeKey: CompositeKey = new CompositeKey([pkv]);
-    // //create view browser component - this will be used to display views
-    // //then create a new component for applications that wraps around the view browser component 
-    // let loadResult = await folderEntity.InnerLoad(compositeKey);
-    // if(!loadResult){
-    //   this.sharedService.CreateSimpleNotification(`Unable to fetch folder ${folder.Name}`, "error", 3500);
-    //   this.showLoader = false;
-    //   return false;
-    // }
+    const categoryEntity = folder.CategoryEntity;
+    const deleteResult: boolean = await categoryEntity.Delete();
+    if(!deleteResult){
+      LogError(`Unable to delete folder ${folder.Name}`, undefined, categoryEntity.LatestResult);
+      this.sharedService.CreateSimpleNotification(`unable to delete folder ${folder.Name}.`, "error", 2500);
+      return false;
+    }
 
-    // let deleteResult = await folderEntity.Delete();
-    // if(!deleteResult){
-    //   this.sharedService.CreateSimpleNotification(`Unable to delete folder ${folder.Name}`, "error", 3500);
-    //   this.showLoader = false;
-    //   return false;
-    // }
-    // else{
-    //   this.sharedService.CreateSimpleNotification(`Successfully deleted folder ${folder.Name}`, "info", 2000);
-    //   this.showLoader = false;
-    // }
-    // return true;
+    this.sharedService.CreateSimpleNotification(`Successfully deleted folder ${folder.Name}`, "info", 2500);
+    return true;
   }
 
   private async deleteResource(item: Item): Promise<boolean> {
+    let genericEntity: BaseEntity = <BaseEntity>item.Data;
+    if(!genericEntity){
+      return false;
+    }
 
     return true;
```

Comment:
@JS-BC  why was this all removed?

Reply:
By having the item.Data field be an entity object instead of a plain object, the amount of logic needed to delete the entity/resource was cut down by a lot